cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/data/ClientConfig.java

```diff
@@ -146,6 +158,16 @@ public boolean clearSnapshot()
         return clearSnapshot;
     }
 
+    public String userProvidedSnapshotTtl()
+    {
+        return userProvidedSnapshotTtl;
+    }
+
+    public String effectiveSnapshotTtl()
+    {
+        return effectiveSnapshotTtl;
+    }
```

Comment:
It looks like the `effectiveSnapshotTtl` is the resolved value on reading from the map. It is confusing to expose both the raw and the resolved values. 
Having `effectiveSnapshotTtl()` is suffice.

Reply:
The user provided option is used for logging:
```
 if (!options.clearSnapshot() && options.userProvidedSnapshotTtl() != null)
```